chia/consensus/blockchain.py

```diff
@@ -988,6 +988,11 @@ async def get_block_records_at(self, heights: List[uint32], batch_size: int = 90
             records.extend(res)
         return records
 
+    def try_block_record(self, header_hash: bytes32) -> Optional[BlockRecord]:
+        if self.contains_block(header_hash):
+            return self.block_record(header_hash)
```

Comment:
Is it worth the extra function call to pre-check vs. just 'translating' the exception into a `None` return?  I don't know how performance sensitive this is nor what the hit vs. miss ratio is.  Could even just be `return self.__whatever_cache.get(header_hash)`.

I'm not set on any change here.  Just sharing some thoughts.

Reply:
I doubt there's any performance value in having both of these. I suspect it's mostly about using these calls in places where we always expect the block to exist (and consider the key error as fatal) or where we expect it to sometimes not exist.